src/binaryutils.cpp

```diff
@@ -232,22 +234,31 @@ static Status validateTensor(const std::shared_ptr<TensorInfo>& tensorInfo,
         return StatusCode::INVALID_SHAPE;
     }
 
-    if (checkBatchSizeMismatch(tensorInfo, src.contents().bytes_contents_size())) {
+    size_t batchSize = !rawInputsContentsUsed ? src.contents().bytes_contents_size() : 1;
+    if (checkBatchSizeMismatch(tensorInfo, batchSize)) {
         SPDLOG_DEBUG("Input: {} request batch size is incorrect. Expected: {} Actual: {}",
             tensorInfo->getMappedName(),
             tensorInfo->getBatchSize().has_value() ? tensorInfo->getBatchSize().value().toString() : std::string{"none"},
             src.contents().bytes_contents_size());
         return StatusCode::INVALID_BATCH_SIZE;
     }
 
-    for (int i = 0; i < src.contents().bytes_contents_size(); i++) {
-        if (src.contents().bytes_contents(i).size() <= 0) {
-            return StatusCode::BYTES_CONTENTS_EMPTY;
-        }
+    if (batchSize <= 0) {
```

Comment:
Shouldnt this if be above `checkBatchSizeMismatch`? 

`checkBatchSizeMismatch` checks if metadata would accept the actual static `batchSize` from request. Passing 0 would collide with `any` (0) in metadata.